include/deal.II/base/parameter_handler.h

```diff
@@ -1127,6 +1127,25 @@ class ParameterHandler : public Subscriptor
                 const std::string           &documentation = "",
                 const bool                   has_to_be_set = false);
 
+  /**
+   * Mark a previously declared parameter as deprecated. This will cause an
+   * exception of type ExcEncounteredDeprecatedEntries to be thrown if
+   * the parameter is used in the input file.
+   *
+   * The exception message will list the name of the parameter.
+   *
+   * @param entry The name of the parameter to mark as deprecated.
+   * @param deprecation_status An optional parameter that can be used to
+   *  toggle the deprecation status of the parameter. If set to true, the
+   *  parameter will be marked as deprecated. This is the default behavior.
+   *  If set to false, the parameter will be marked as not deprecated.
+   *  This is useful if a parameter is marked as deprecated by
+   *  one function, but another function wants to keep using it.
+   */
+  void
+  mark_as_deprecated(const std::string &entry,
+                     const bool         deprecation_status = true);
```

Comment:
Perhaps call this variable `is_deprecated`? `deprecation_status` as a boolean could be interpreted either way.